dash-renderer/src/checkPropTypes.js

```diff
@@ -0,0 +1,89 @@
+ * @return {string} Any error messsage resulting from checking the types
+ */
+export default function checkPropTypes(
+    typeSpecs,
+    values,
+    location,
+    componentName,
+    getStack
+) {
+    const errors = [];
+    for (const typeSpecName in typeSpecs) {
+        if (typeSpecs.hasOwnProperty(typeSpecName)) {
+            let error;
+            // Prop type validation may throw. In case they do, we don't want to
+            // fail the render phase where it didn't fail before. So we log it.
+            // After these have been cleaned up, we'll let them throw.
+            try {
+                // This is intentionally an invariant that gets caught. It's the same
+                // behavior as without this statement except with a better message.
+                if (typeof typeSpecs[typeSpecName] !== 'function') {
+                    error = Error(
+                        (componentName || 'React class') +
+                            ': ' +
+                            location +
+                            ' type `' +
+                            typeSpecName +
+                            '` is invalid; ' +
+                            'it must be a function, usually from the `prop-types` package, but received `' +
+                            typeof typeSpecs[typeSpecName] +
+                            '`.'
+                    );
+                    error.name = 'Invariant Violation';
+                } else {
+                    error = typeSpecs[typeSpecName](
+                        values,
+                        typeSpecName,
+                        componentName,
+                        location,
+                        null,
+                        ReactPropTypesSecret
+                    );
+                }
+            } catch (ex) {
+                error = ex;
+            }
+            if (error && !(error instanceof Error)) {
+                errors.push(
+                    (componentName || 'React class') +
+                        ': type specification of ' +
+                        location +
+                        ' `' +
+                        typeSpecName +
+                        '` is invalid; the type checker ' +
+                        'function must return `null` or an `Error` but returned a ' +
+                        typeof error +
+                        '. ' +
+                        'You may have forgotten to pass an argument to the type checker ' +
+                        'creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and ' +
+                        'shape all require an argument).'
+                );
+            }
+            if (error instanceof Error) {
+                var stack = (getStack && getStack()) || '';
+
+                errors.push(
+                    'Failed ' + location + ' type: ' + error.message + stack
+                );
+            }
+        }
+    }
+    return errors.join('\n\n');
+}
```

Comment:
Can you explain the additional changes that have been made to make it `simplified and tweaked to our needs`. Arguably, the source file hasn't been changed in a long time either and it might be no worse to have our own copy than it is to depend on someone else's copy.. 

Since the original file is under a BSD license, we should make sure that we either retain the original disclaimer as done in check-prop-type (https://github.com/ratehub/check-prop-types/blob/master/index.js#L7) or make sure the disclaimer for React is included explicitly in our bundled code -- which it wouldn't right now as we externalize React dependencies.

Reply:
Current `prop-types` [license is MIT](https://github.com/facebook/prop-types/blob/master/LICENSE) (it was changed from BSD shortly after check-prop-types was published), as is [check-prop-types](https://github.com/ratehub/check-prop-types/blob/master/LICENSE) - that presumably means this concern is moot, yes?

Reply:
Moot as moot can be